/**:
  ros__parameters:
    start_planner:

      th_arrived_distance: 1.0
      th_stopped_velocity: 0.01
      th_stopped_time: 1.0
      collision_check_margins: [2.0, 1.0, 0.5, 0.1]
      collision_check_margin_from_front_object: 5.0
      skip_rear_vehicle_check: false
      extra_width_margin_for_rear_obstacle: 0.5
      th_moving_object_velocity: 1.0
      object_types_to_check_for_path_generation:
        check_car: true
        check_truck: true
        check_bus: true
        check_trailer: true
        check_bicycle: true
        check_motorcycle: true
        check_pedestrian: true
        check_unknown: true
      th_distance_to_middle_of_the_road: 0.5
      center_line_path_interval: 1.0
      # shift pull out
      enable_shift_pull_out: true
      check_shift_path_lane_departure: true
      allow_check_shift_path_lane_departure_override: false
      shift_collision_check_distance_from_end: -10.0
      minimum_shift_pull_out_distance: 0.0
      maximum_longitudinal_deviation: 1.0 # Note, should be the same or less than planning validator's "longitudinal_distance_deviation"
      lateral_jerk: 0.5
      lateral_acceleration_sampling_num: 3
      minimum_lateral_acc: 0.15
      maximum_lateral_acc: 0.5
      maximum_curvature: 0.07
      end_pose_curvature_threshold: 0.1
      # geometric pull out
      enable_geometric_pull_out: true
      geometric_collision_check_distance_from_end: 0.0
      arc_path_interval: 1.0
      divide_pull_out_path: true
      geometric_pull_out_velocity: 1.0
      lane_departure_margin: 0.2
      lane_departure_check_expansion_margin: 0.0
      backward_velocity: -1.0
      pull_out_max_steer_angle: 0.26  # 15deg
      # search start pose backward
      enable_back: true
      search_priority: "efficient_path"  # "efficient_path" or "short_back_distance"
      max_back_distance: 20.0
      backward_search_resolution: 2.0
      backward_path_update_duration: 3.0
      ignore_distance_from_lane_end: 0.0
      # turns signal
      prepare_time_before_start: 0.0
      # freespace planner
      freespace_planner:
        enable_freespace_planner: true
        end_pose_search_start_distance: 20.0
        end_pose_search_end_distance: 30.0
        end_pose_search_interval: 2.0
        freespace_planner_algorithm: "astar"  # options: astar, rrtstar
        velocity: 1.0
        vehicle_shape_margin: 1.0
        time_limit: 3000.0
        max_turning_ratio: 0.7
        turning_steps: 1
        # search configs
        search_configs:
          theta_size: 120
          angle_goal_range: 6.0
          lateral_goal_range: 0.5
          longitudinal_goal_range: 2.0
          curve_weight: 0.5
          reverse_weight: 1.0
          direction_change_weight: 1.5
        # costmap configs
        costmap_configs:
          obstacle_threshold: 30
        # -- A* search Configurations --
        astar:
          search_method: "forward"  # options: forward, backward
          only_behind_solutions: false
          use_back: true
          adapt_expansion_distance: true
          expansion_distance: 0.5
          near_goal_distance: 3.0
          distance_heuristic_weight: 2.0
          smoothness_weight: 0.5
          obstacle_distance_weight: 1.75
          goal_lat_distance_weight: 5.0
        # -- RRT* search Configurations --
        rrtstar:
          enable_update: true
          use_informed_sampling: true
          max_planning_time: 150.0
          neighbor_radius: 8.0
          margin: 1.0

      stop_condition:
        maximum_deceleration_for_stop: 1.0
        maximum_jerk_for_stop: 1.0
      path_safety_check:
        # EgoPredictedPath
        ego_predicted_path:
          min_velocity: 0.0
          min_acceleration: 1.0
          time_horizon_for_front_object: 10.0
          time_horizon_for_rear_object: 10.0
          time_resolution: 0.5
          delay_until_departure: 1.0
        # For target object filtering
        target_filtering:
          safety_check_time_horizon: 10.0
          safety_check_time_resolution: 1.0
          # detection range
          object_check_forward_distance: 10.0
          object_check_backward_distance: 100.0
          ignore_object_velocity_threshold: 1.0
          # ObjectTypesToCheck
          object_types_to_check:
            check_car: true
            check_truck: true
            check_bus: true
            check_trailer: true
            check_bicycle: true
            check_motorcycle: true
            check_pedestrian: true
            check_unknown: false
          # ObjectLaneConfiguration
          object_lane_configuration:
            check_current_lane: true
            check_right_side_lane: true
            check_left_side_lane: true
            check_shoulder_lane: true
            check_other_lane: false
          include_opposite_lane: false
          invert_opposite_lane: false
          check_all_predicted_path: true
          use_all_predicted_path: true
          use_predicted_path_outside_lanelet: false

        # For safety check
        safety_check_params:
          # safety check configuration
          enable_safety_check: true
          # collision check parameters
          publish_debug_marker: false
          rss_params:
            rear_vehicle_reaction_time: 2.0
            rear_vehicle_safety_time_margin: 1.0
            lateral_distance_max_threshold: 2.0
            longitudinal_distance_min_threshold: 3.0
            longitudinal_velocity_delta_time: 0.8
            extended_polygon_policy: "along_path" # [-] select "rectangle" or "along_path"
          # hysteresis factor to expand/shrink polygon
          hysteresis_factor_expand_rate: 1.0
          collision_check_yaw_diff_threshold: 1.578
          # temporary
          backward_path_length: 30.0
          forward_path_length: 100.0

      surround_moving_obstacle_check:
          search_radius: 10.0
          th_moving_obstacle_velocity: 1.0
          # ObjectTypesToCheck
          object_types_to_check:
            check_car: true
            check_truck: true
            check_bus: true
            check_trailer: true
            check_bicycle: true
            check_motorcycle: true
            check_pedestrian: true
            check_unknown: false

      # debug
      debug:
        print_debug_info: false
